src/v/kafka/client/transport.h

```diff
@@ -120,20 +133,20 @@ class transport : public net::base_transport {
     requires(KafkaApi<typename T::api_type>)
       ss::future<typename T::api_type::response_type> dispatch(
         T r, api_version request_version, api_version response_version) {
-        return send_recv(
-                 T::api_type::key,
-                 request_version,
-                 [this, request_version, r = std::move(r)](
-                   response_writer& wr) mutable {
-                     write_header(wr, T::api_type::key, request_version);
-                     r.encode(wr, request_version);
-                 })
-          .then([response_version](iobuf buf) {
-              using response_type = typename T::api_type::response_type;
-              response_type r;
-              r.decode(std::move(buf), response_version);
-              return ss::make_ready_future<response_type>(std::move(r));
+        check_exceptional_cases<T>(request_version);
+        auto buf = co_await send_recv(
+          T::api_type::key,
```

Comment:
coroutines in templates is a source of `fatal error: error in backend: Cannot represent a difference across sections` compiler bug which i think has been fixed but isn't widely available yet in clang

Reply:
I can revert making this method a coroutine and just have `check_exceptional_cases` return an exception future instead